Second CSV parsing fix

dirkhh opened the pull request:
Split into the correct number of fields.
Also, add exception handling.
Again, with theanks to Kat for help with the syntax.

dirkhh committed:
Second CSV parsing fix

Split into the correct number of fields.
Also, add exception handling.
Again, with theanks to Kat for help with the syntax.

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>

iakat commented:
```patch
diff --git a/src/adsb_api/utils/provider.py b/src/adsb_api/utils/provider.py
index 6cedfe8..8732df3 100644
--- a/src/adsb_api/utils/provider.py
+++ b/src/adsb_api/utils/provider.py
@@ -242,11 +242,10 @@ class RedisVRS:
                     # make redis transaction
                     pipeline = self.redis.pipeline()
 
-                    for row in csv.DictReader(data.splitlines()):
-                        values = list(row.values())
-                        key = f"vrs:{name}:{values[0]}"
-                        rest_of_row = ",".join(values[1:])
-                        pipeline = pipeline.set(key, rest_of_row)
+                    for row in data.splitlines():
+                        name = row.split(",")[0]
+                        key = f"vrs:{name}:{name}"
+                        pipeline = pipeline.set(key, row)
                     print("vrsx y", len(pipeline))
                     await pipeline.execute()

```
missing this to make sure the full line is set in redis

dirkhh committed:
Include key in cached Redis data

This way what Redis returns matches the CSV data one would read directly
from the file.

Suggested-by: Katia <katia@linux.com>
Signed-off-by: Dirk Hohndel <dirk@hohndel.org>
